### ISSUE TITLE
Could not connect to server.

### ISSUE THREAD

**Issue by asbjornenge** [LN-Zap/zap-iOS#23]
**Describe the bug**
Unable to connect to lnd node even though everything should be set up correctly.

* lnd is up and running with all the correct params. 
* synced to chain. 
* wallet unIocked. 
* I can `telnet <ip> 10009` no problem :man_shrugging:
* certs look correct
* zapconnect worked perfectly

Seems like it's not even trying, I get the error immediately.

**Expected behavior**
Should be able to connect 😛 

**Your environment**
- Device: iPhone6s
- OS: iOS 11.2.5
- Version: latest from TestFlight

**Comment by ottosuess** [comment]
are you using `rpclisten=0.0.0.0:10009`? 🤔

a good way to debug this is trying to run a local `lncli` to control your remote node.
by running `lncli --rpcserver=ip:10009 --tlscertpath=... --macaroonpath=... getinfo`.

if that's working the app should be able to connect to your server too.

**Comment by asbjornenge** [comment]
@ottosuess aha! Thanks for a good debug tip 😉 

```
[lncli] rpc error: code = Internal desc = connection error: desc = "transport: authentication handshake failed: x509: certificate is valid for 127.0.0.1, ::1, 172.17.0.13, fe80::42:acff:fe11:d, not <external_ip>"
```

I'm assuming `zapconnect` is generating the certs? Is there a way to specify the `ip` to generate the cert for? I'm running lnd in a container (ref. convo on twitter where I'd document it for your guys 😉) which might be why this is not working as expected. It got the external ip right for the QR code config that goes into the app though 🤔

**Comment by asbjornenge** [comment]
FYI: `172.17.0.13` is the ip given to the container by docker (eth0 interface inside the container). If at all possible it would be great being able to specify the ip used by zapconnect 👍

**Comment by asbjornenge** [comment]
Ok, so I figured out it was `lnd` creating the cert and it has a `--tlsextraip` parameter 😬🎉

It's working now 💪 I'll try to get to that tutorial during this week and ship you guys a PR with Dockerfile + docs 👍 

Thanks for awesomesauce software ✨

**Comment by dennisreimann** [comment]
For those of you using `rpclisten=0.0.0.0:10009` and encountering similar trouble, see this issue as a reference: lightningnetwork/lnd#1567

TL;DR: Also adding the `tlsextraip=0.0.0.0` config helps.
